static/js/popovers.js

```diff
@@ -547,6 +547,14 @@ exports.register_click_handlers = function () {
         e.stopPropagation();
     });
 
+    $('body').on('click', '.mark_stream_as_read', function (e) {
+        var stream = $(e.currentTarget).parents('ul').attr('data-name');
+        //var topic = $(e.currentTarget).attr('data-topic-name');
```

Comment:
Okay, awesome, thanks for the info. I'm finishing up the error handling in the messages.py and the extra test case then I'll give this a shot.